### PR TITLE
feat: Initialize Android support

### PR DESCRIPTION
**Pull request by Cryolitia**

### CODE REVIEW

File: config/DepsConfig.json [hunk 1]
```diff
@@ -14,7 +14,9 @@
         "url": "https://github.com/nodejs-mobile/nodejs-mobile/archive/refs/tags/v18.17.2.tar.gz",
         "filename": "nodejs-mobile-v18.17.2.tar.gz",
         "ios-binary-url": "https://github.com/nodejs-mobile/nodejs-mobile/releases/download/v18.17.2/nodejs-mobile-v18.17.2-ios.zip",
-        "ios-binary-filename": "nodejs-mobile-v18.17.2-ios.zip"
+        "ios-binary-filename": "nodejs-mobile-v18.17.2-ios.zip",
+        "android-binary-url": "https://github.com/nodejs-mobile/nodejs-mobile/releases/download/v18.17.2/nodejs-mobile-v18.17.2-android.zip",
+        "android-binary-filename": "nodejs-mobile-v18.17.2-ios.zip",
```
Comment (harry75369) [inline config/DepsConfig.json:19]: a typo?

File: README.md [hunk 2]
```diff
@@ -106,6 +106,22 @@ cmake --build . --parallel -t vgg_container
...
+We currently only support Android NDK = 27 and Android API >= 24.
+
+You need to download Android NDK r27 from here: https://github.com/android/ndk/releases/tag/r27
+
+Build & install `vgg_container` libraries for [vgg_android](https://github.com/verygoodgraphics/vgg_android).
+
+```bash
+mkdir build.android
+cd build.android
+cmake .. -DCMAKE_TOOLCHAIN_FILE=<path_to_android_ndk>/android-ndk-r27/build/cmake/android.toolchain.cmake -DVGG_VAR_TARGET="Android-x86_64" -DANDROID_NDK=<path_to_android_ndk>/android-ndk-r27
```
Comment (harry75369) [inline README.md]: Maybe the explicit directory name `android-ndk-r27` could be removed? We could assume `<path_to_android_ndk>` already includes it.

**Comment by Cryolitia** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cmake/SkiaUtils.cmake [hunk 3]
```diff
@@ -208,6 +210,20 @@ skia_use_system_zlib=false
...
+skia_use_system_freetype2=false
+skia_use_system_libpng=false
+skia_use_system_libjpeg_turbo=false
+skia_use_system_harfbuzz=false
+skia_use_system_libwebp=false
+skia_use_expat=true
+skia_use_freetype=true
+skia_use_system_expat=false
+skia_use_system_icu=false
+skia_use_vulkan=true")
```
Comment (harry75369) [inline cmake/SkiaUtils.cmake:225]: Unless explicitly disabled by setting `skia_use_vulkan=false` as emscripten build does, whether turning on vulkan or not should depend on `VULKAN_VARIABLE` likes other builds.

And this also adds a bit flexibility if we want to lower the api version limit some day.

File: README.md [hunk 4]
```diff
@@ -106,6 +106,22 @@ cmake --build . --parallel -t vgg_container
...
+We currently only support Android NDK = 27 and Android API >= 24.
+
+You need to download Android NDK r27 from here: https://github.com/android/ndk/releases/tag/r27
+
+Build & install `vgg_container` libraries for [vgg_android](https://github.com/verygoodgraphics/vgg_android).
+
+```bash
+mkdir build.android
+cd build.android
+cmake .. -DCMAKE_TOOLCHAIN_FILE=<path_to_android_ndk>/android-ndk-r27/build/cmake/android.toolchain.cmake -DVGG_VAR_TARGET="Android-x86_64" -DANDROID_NDK=<path_to_android_ndk>/android-ndk-r27 -DANDROID_PLATFORM=android-24
```
Comment (harry75369) [inline README.md:120]: The explicit directory name `android-ndk-r27` could be removed as `<path_to_android_ndk>` already includes it right?

File: lib/CMakeLists.txt [hunk 5]
```diff
@@ -85,13 +86,21 @@ if(VGG_VAR_TARGET MATCHES "^iOS")
     file(RENAME ${NODEJS_MOBILE_DIR} "${CMAKE_CURRENT_SOURCE_DIR}/${NODE_FOLDER}")
   endif()
 
-  # Download the pre-built binary (NodeMobile.xcframework)
-  set(LIB_NODE_FOLDER "nodejs-mobile-ios" CACHE STRING "lib node folder name" FORCE)
+  # Download the iOS pre-built binary (NodeMobile.xcframework)
+  set(LIB_NODE_FOLDER "nodejs-mobile-lib" CACHE STRING "lib node folder name" FORCE)
```
Comment (harry75369) [inline lib/CMakeLists.txt:91]: Please rename the folder name `nodejs-mobile-ios` in `src/Entry/Container/CMakeLists.txt` as well

File: cmake/SkiaUtils.cmake [hunk 6]
```diff
@@ -1,14 +1,28 @@
...
+  AND NOT VGG_VAR_TARGET MATCHES "^Android")
   find_package(Vulkan)
 endif()
+
 if(Vulkan_FOUND)
 set(VULKAN_AVAILABLE "true")
 else()
 set(VULKAN_AVAILABLE "false")
 endif()
+if (VGG_VAR_TARGET MATCHES "^Android")
```
Comment (harry75369) [inline cmake/SkiaUtils.cmake:15]: Better move the above `Vulkan_FOUND` handling logic into the else branch.

**Pull request merged**